### PR TITLE
add button to hide tiles

### LINKED ISSUE #2050: Feature request: a button to hide tiles
A button next to the main game board. Once toggle, it makes all tiles 25% opacity so that users can look at what are underneath.

### PR DESCRIPTION
**Pull request by vsrisuknimit**
Normal image

Once click 'hide tiles'
image

It does not persist. As soon as you refresh or other player take an action, the view mode goes back to default. I think this is what we want.

Question: Not sure what the button name should be. 'hide tiles'? 'show bonus'?
Question: Should we show player cube during hide tiles?

Still have to adjust the position so it does show over the card during initial research phase.

This PR can close #2050 .

### CODE REVIEW

**Comment by vsrisuknimit** [comment]
@chosta help. I need the position of the button to be absolute so it does shift the terraforming tracks. But during initial draft, absolute just layover cards. What can I do?

image

File: src/components/Board.ts [hunk 1]
```diff
@@ -157,9 +158,16 @@ export const Board = Vue.component('board', {
...
+    hideTile: function() {
+      return this.isTileHidden = !this.isTileHidden;
+    },
+    checkHideTile: function():boolean {
+      return this.isTileHidden;
+    },
   },
   template: `
     <div :class="getGameBoardClassName()">
+        <div class="hide-tile-button" v-on:click.prevent="hideTile()">Hide tiles</div>
```
Comment (kberg) [inline src/components/Board.ts:170]: "show map"?

**Review by kberg — COMMENTED** [review]
IMO an excellent bit of work. While there are more elegant solutions, this is one of those cases where the perfect would be the enemy of the good.

How would you envision this applying to The Moon simultaneously? I wouldn't worry about this just yet, though, but I would like your opinion.

File: src/components/Board.ts (hunk 1 above)
Comment (chosta) [reply src/components/Board.ts:170]: "show tiles"?

File: src/components/Board.ts [hunk 2]
```diff
@@ -157,9 +158,16 @@ export const Board = Vue.component('board', {
     getGameBoardClassName: function():string {
       return this.venusNextExtension ? 'board-cont board-with-venus' : 'board-cont board-without-venus';
     },
+    hideTile: function() {
```
Comment (chosta) [inline src/components/Board.ts:161]: Devs like the word `toggle` in such cases.

File: src/components/Board.ts [hunk 3]
```diff
@@ -157,9 +158,16 @@ export const Board = Vue.component('board', {
...
+      return this.isTileHidden = !this.isTileHidden;
```
Comment (chosta) [inline src/components/Board.ts:162]: I'd prefer to have this in two lines so we separate the concerns.
1) Assignment
2) return statement

**Review by chosta — CHANGES_REQUESTED** [review]
If you want to have something that is positioned absolutely relative to the parent container, just add "position: relative" (to the parent)

**Comment by chosta** [comment]
Also, you have to toggle the text too - Hide tiles should become Show tiles and vice versa (upon click)

File: src/styles/board.less [hunk 4]
```diff
@@ -284,3 +284,18 @@
     margin-bottom: -1px;
 }
 
+.board-hidden-tile{
+	opacity: 0.6;
+}
+
+.hide-tile-button {
+	position: absolute;
+	top: 610px;
```
Comment (chosta) [inline src/styles/board.less:293]: Better to offset it from "bottom"

File: src/styles/board.less [hunk 5]
```diff
@@ -284,3 +284,18 @@
...
+	text-transform: uppercase;
+	border-radius: 4px;
+	font-size: 12px;
```
Comment (chosta) [inline src/styles/board.less:296]: Please, use a variable for the font size from "variables.less"

File: src/styles/board.less [hunk 6]
```diff
@@ -284,3 +284,18 @@
     margin-bottom: -1px;
 }
 
+.board-hidden-tile{
+	opacity: 0.6;
+}
+
+.hide-tile-button {
```
Comment (chosta) [inline src/styles/board.less:291]: Please, consider reusing a class that has been used previously. Check out the buttons for "concise" and "full" view in the player board.

File: src/components/Board.ts (hunk 2 above)
Comment (vsrisuknimit) [reply src/components/Board.ts:161]: @chosta if I use relative, all the global parameters will be pushed aside.

File: src/components/Board.ts (hunk 3 above)
Comment (vsrisuknimit) [reply src/components/Board.ts:162]: I guess it does not need to return anything. my bad.

File: src/styles/board.less (hunk 6 above)
Comment (vsrisuknimit) [reply src/styles/board.less:291]: That class is inside another class. I cannot use it from here.

**Comment by vsrisuknimit** [comment]
Before
image

After
image

File: src/styles/board.less (hunk 5 above)
Comment (vsrisuknimit) [reply src/styles/board.less:296]: use font small now

File: src/styles/board.less (hunk 4 above)
Comment (vsrisuknimit) [reply src/styles/board.less:293]: bottom now. with pseudoelement to be relative.

File: src/components/Board.ts (hunk 2 above)
Comment (vsrisuknimit) [reply src/components/Board.ts:161]: changed the function name

File: src/components/Board.ts (hunk 1 above)
Comment (vsrisuknimit) [reply src/components/Board.ts:170]: now show tiles and hide tile

**Comment by vsrisuknimit** [comment]
> How would you envision this applying to The Moon simultaneously? I wouldn't worry about this just yet, though, but I would like your opinion.

@kberg I imagine the same button will hide tiles on the Moon map as well. It should be pretty easy to add.

**Comment by vsrisuknimit** [comment]
Do we want to hide player cube as well?

image

@chosta I would like to apply translucent only to the tile but not the bonuses. Is there anyway to do it?

File: src/styles/board.less (hunk 6 above)
Comment (chosta) [reply src/styles/board.less]: My point was to use the style of an existing UI element for consistency. Plase, consider:
`    text-transform: uppercase;
    border-radius: 4px;
    color: black;
    background: #ccc;`

**Review by chosta — APPROVED** [review]
I don't think there is a way (not an easy one at least) to have an element without opacity inside another that is opaque. Just leave it as it is, it does the job and it's just fine.
Check my comment about styling the button before merge.

File: src/styles/board.less (hunk 6 above)
Comment (vsrisuknimit) [reply src/styles/board.less]: I have all those styles here as well except the background color. I don't want this button to be too bright next to the main map.

**Comment by vsrisuknimit** [comment]
Two changes:
- Add transparency only to tiles, not bonuses
- Remove player cubes when hiding tiles

Before:
image

After:
image

**Review by chosta — APPROVED** [review]
Yeah, if it's inside a container you can stack opacities. Good work.

**Pull request merged**